Is support for FreeBSD planned?

Currently build fails on FreeBSD:
```
error[E0599]: no method named `st_mode` found for struct `std::fs::Metadata` in the current scope
   --> oxish/src/authentication.rs:727:16
    |
727 |     match meta.st_mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_uid() == uid) {
    |                ^^^^^^^
    |
   --> /rustc/8bab26f4f68e0e26f0bb7960be334d5b520ea452/library/std/src/os/freebsd/fs.rs:33:7
    |
    = note: the method is available for `std::fs::Metadata` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_mode` is implemented but not in scope; perhaps you want to import it
    |
  1 + use std::os::freebsd::fs::MetadataExt;
    |
help: there is a method `mode` with a similar name
    |
727 -     match meta.st_mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_uid() == uid) {
727 +     match meta.mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_uid() == uid) {
    |

error[E0599]: no method named `st_uid` found for struct `std::fs::Metadata` in the current scope
   --> oxish/src/authentication.rs:727:48
    |
727 |     match meta.st_mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_uid() == uid) {
    |                                                ^^^^^^
    |
   --> /rustc/8bab26f4f68e0e26f0bb7960be334d5b520ea452/library/std/src/os/freebsd/fs.rs:37:7
    |
    = note: the method is available for `std::fs::Metadata` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_uid` is implemented but not in scope; perhaps you want to import it
    |
  1 + use std::os::freebsd::fs::MetadataExt;
    |
help: there is a method `st_gid` with a similar name
    |
727 -     match meta.st_mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_uid() == uid) {
727 +     match meta.st_mode() & 0o022 == 0 && (meta.st_gid() == 0 || meta.st_uid() == uid) {
    |

error[E0599]: no method named `st_uid` found for struct `std::fs::Metadata` in the current scope
   --> oxish/src/authentication.rs:727:70
    |
727 |     match meta.st_mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_uid() == uid) {
    |                                                                      ^^^^^^
    |
   --> /rustc/8bab26f4f68e0e26f0bb7960be334d5b520ea452/library/std/src/os/freebsd/fs.rs:37:7
    |
    = note: the method is available for `std::fs::Metadata` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_uid` is implemented but not in scope; perhaps you want to import it
    |
  1 + use std::os::freebsd::fs::MetadataExt;
    |
help: there is a method `st_gid` with a similar name
    |
727 -     match meta.st_mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_uid() == uid) {
727 +     match meta.st_mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_gid() == uid) {
    |

warning: unused import: `std::os::unix::fs::MetadataExt`
  --> oxish/src/authentication.rs:13:5
   |
13 | use std::os::unix::fs::MetadataExt;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default

For more information about this error, try `rustc --explain E0599`.
warning: `oxish` (lib) generated 1 warning
error: could not compile `oxish` (lib) due to 3 previous errors; 1 warning emitted
```

The issue is that `std::os::unix::fs::MetadataExt` doesn't provide `st_mode`/`st_uid` on FreeBSD, those live in `std::os::freebsd::fs::MetadataExt` instead (per the compiler's own suggestion above).

If FreeBSD is a supported/intended target, it should at minimum compile cleanly there. Happy to submit a patch for this if that's welcome, let me know.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -28,6 +28,16 @@ jobs:
       - run: cargo check -p oxish --no-default-features --features aws-lc
       - run: cargo test --features graviola,aws-lc -- --include-ignored
 
+  freebsd:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v7
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          targets: x86_64-unknown-freebsd
+      - run: cargo check --target x86_64-unknown-freebsd -p oxish
+
   msrv:
     runs-on: ubuntu-latest
     steps:

--- a/oxish/src/authentication.rs
+++ b/oxish/src/authentication.rs
@@ -5,11 +5,7 @@ use core::{
     ops::{ControlFlow, Deref},
     time::Duration,
 };
-#[cfg(target_vendor = "apple")]
-use std::os::darwin::fs::MetadataExt;
-#[cfg(target_os = "linux")]
-use std::os::linux::fs::MetadataExt;
-#[cfg(all(unix, not(target_vendor = "apple"), not(target_os = "linux")))]
+#[cfg(unix)]
 use std::os::unix::fs::MetadataExt;
 use std::{
     borrow::Cow,
@@ -724,7 +720,7 @@ fn check_permissions(file: &File, uid: u32, level: &str) -> ControlFlow<()> {
         }
     };
 
-    match meta.st_mode() & 0o022 == 0 && (meta.st_uid() == 0 || meta.st_uid() == uid) {
+    match meta.mode() & 0o022 == 0 && (meta.uid() == 0 || meta.uid() == uid) {
         true => ControlFlow::Continue(()),
         false => ControlFlow::Break(()),
     }

